app/util/api_version_handler.py

```diff
@@ -0,0 +1,78 @@
+import re
+
+
+class APIVersionHandler(object):
+    # List of all versions of the API.
+    _versions = [
+        1,
+        2
+    ]
+
+    @classmethod
+    def resolve_version(cls, header: str, uri: str):
+        """
+        Gets the respective version of the API relative to the request header and URI.
+        :param header: The value of the header which to search for version type (Content-Type/Accept).
+        :type header: str
+        :param uri: The URI from the request being checked.
+        :type uri: str
+        :rtype: int
+        """
+        matches = re.match(r'.*(?:application/vnd.clusterrunner.v(\d+)\+json).*', header, re.I)
```

Comment:
Having `get_first` and `get_last` can be useful in other places where we want to know the first and last version of the API (might not be as relevant for `get_first`, but I know that we definitely want to use `get_last` in other spots later down the line). I don't mind the idea of condensing some of these functions though, especially the internal methods